open_dread_rando/door_patcher.py

```diff
@@ -1,59 +1,309 @@
+from enum import Enum
+from pathlib import Path
+
+from construct import Container, ListContainer
+from mercury_engine_data_structures.formats.bmmap import Bmmap
 from open_dread_rando.patcher_editor import PatcherEditor
 
+# copied from existing entity so we don't have to make a whole shield
+_EXAMPLE_SHIELD = {"scenario": "s010_cave", "layer": "default", "actor": "Door003_missileShield"}
+# this one is just copied because *something* weird kills you whenever you touch it otherwise. 
+_EXAMPLE_PHANTOM_DOOR = {"scenario": "s010_cave", "layer": "default", "actor": "Door049 (PR-PR)"}
+
+"""
+Enum representing minimap data. Also contains functions that return mapDoor and mapBlockage Containers. 
+
+iconId: a nondirectional icon id. 'L' or 'R' is appended for each direction. 
+tuple: the offsets for the oBoxes. All values should be taken from a "left" entity:
+    0: x-axis offset from vPos to the bottom-right corner
+    1: x-axis offset from vPos to the top-left corner
+    2: y-axis offset from vPos to the bottom-right corner
+    3: y-axis offset from vPos to the top-left corner
+"""
+class MinimapIconData(Enum):
+    INVALID = ("INVALID", (0, 0, 0, 0))
```

Comment:
there's no need for an `INVALID` value in the enums. any time you'd use it, just raise an exception immediately

Reply:
Is there any preferred exception type? My guess is `ValueError` but i'm not sure if we have any custom exceptions in the project that would be a better fit.